pkg/workceptor/remote_work.go

```diff
@@ -535,6 +536,7 @@ func (rw *remoteUnit) Restart() error {
 	if red.RemoteStarted {
 		return rw.startOrRestart(false)
 	}
+	rw.topJC = nil
```

Comment:
`return rw.startOrRestart(false)` may also return an error, and we might need to set topJC to nil in that case too